### PR TITLE
Add support for template versioning in the application management API

### PR DESCRIPTION
**Pull request by DilshanSenarath**
### Proposed changes in this pull request

- Add support for specifying the template version when creating an application
- Add support for updating the template version of an application
- Add support for retrieving the `templateVersion` when fetching the specified service provider data
- Add support for requesting the `templateVersion` as a `requiredAttribute` in application listing data

### Related Issues

- https://github.com/wso2/product-is/issues/20108

### CODE REVIEW

**Review by ImalshaG — DISMISSED** [review]

**Comment by DilshanSenarath** [comment]
The PR builder is failing because we need to update the integration tests on the product-is side. These changes will be merged through https://github.com/wso2/product-is/pull/20745.

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/application/management/v1/ApplicationListItem.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
```
Comment (pavinduLakshan) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/application/management/v1/ApplicationListItem.java:2]: Let's fix the other places as well.

```suggestion
 * Copyright (c) 2023-2024, WSO2 LLC. (http://www.wso2.com).
```

Comment (DilshanSenarath) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/application/management/v1/ApplicationListItem.java:2]: This class is generated through the API generator, which handles adding headings and other updates. Typically, we do not modify these generated contents.

Comment (pavinduLakshan) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/application/management/v1/ApplicationListItem.java:2]: Ack, my bad i didn't see it was inside gen.

Comment (dhaura) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/application/management/v1/ApplicationListItem.java:2]: Going forward as our current practice, we use the year range instead.

Comment (DilshanSenarath) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/application/management/v1/ApplicationListItem.java:2]: Ack. Will change the copyright. Thanks @dhaura and @pavinduLakshan

Comment (DilshanSenarath) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/application/management/v1/ApplicationListItem.java]: https://github.com/wso2/identity-api-server/pull/634/commits/f6901119a7b6990285779cef13f26dd0fd1df87c

**Review by dhaura — APPROVED** [review]

**Pull request merged**